File: LeetCode1--两数之和/2.js
var twoSum = function (arr, res) {
    var diffs = {};
    var len = arr.length;
    for (var i = 0; i < len; i++) {
        if (diffs[res - arr[i]] !== undefined) {
            return [diffs[res - arr[i]], i];
        }
        diffs[arr[i]] = i;
    }
};

File: LeetCode1200--最小绝对差/2.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        int minDiff = INT_MAX;
        vector<vector<int>> res;
        // 找出最小差值
        for (int i = 1; i < arr.size(); i++) {
            minDiff = min(minDiff, arr[i] - arr[i - 1]);
        }
        // 收集结果
        for (int i = 1; i < arr.size(); i++) {
            if (arr[i] - arr[i - 1] == minDiff) {
                res.push_back({arr[i - 1], arr[i]});
            }
        }
        return res;
    }
};